src/universal/utils/fromNow.js

```diff
@@ -26,3 +26,16 @@ export default function fromNow(time) {
   // this is both for eslint, and for chuckles. It should never happen:
   return 'infinitely long ago';
 }
+
+export function refreshPeriod(time) {
+  const msElapsed = (Date.now() - time) || 0;
+  const threshKeys = Object.keys(thresholds);
+  for (let i = 1; i < threshKeys.length; i++) {
```

Comment:
@jordanh Here is what the forEach solution would look like:
```
export function refreshPeriod(time) {
  const msElapsed = (Date.now() - time) || 0;
  const threshKeys = Object.keys(thresholds);
  let period = null;
  let periodCalculated = false;
  threshKeys.forEach((key, idx) => {
    if (periodCalculated) return;
    if (msElapsed < thresholds[key]) {
      period = key === 'minute' ? 30 * thresholds.second :thresholds[threshKeys[idx - 1]];
      periodCalculated = true;
    }
  });
  if (!period) throw new Error('Infinite timestamp calculated in fromNow.js!');
  return period;
}
```

I think it is better as a traditional for loop because execution of the entire function stops when we return within the for loop and we don't need extra variables for bookkeeping (period & periodCalculated)